File: 04/Assets/4.1/missing_values.py
import pandas as pd
# load the data
headers=['Index', 'Name','DoB','Phone']
data = pd.read_csv('bad_data.csv', header=None, names=headers)
# count the missing values
missing_values = data.isnull().sum()
print(missing_values)
# print the row/s with missing values
print(data[data.isnull().any(axis=1)])

File: 04/Assets/4.1/missing_values.R
# install.packages('dplyr', repos = "http://cran.us.r-project.org")
# load the library, suppressing output
library(dplyr, warn.conflicts = FALSE)
# lset up our headers
headers <- c(
    "Index", "Name",
    "DoB", "Phone"
)
# load data treating missing values as NA
data <- read.csv("bad_data.csv",
    header = FALSE,
    col.names = headers,
    na.strings = c("", "NA")
)
# print the data
print(data)
# show only rows with the NA values
print("Rows with missing values:")
print(data %>% filter(is.na(DoB) | is.na(Phone) | is.na(Name)))
